src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -532,19 +532,11 @@ private bool RequiresEncoding(StringSegment input)
         }
 
         // Encode using MIME encoding
-        private unsafe string EncodeMime(StringSegment input)
+        private string EncodeMime(StringSegment input)
         {
-            fixed (char* chars = input.Buffer)
-            {
-                var byteCount = Encoding.UTF8.GetByteCount(chars + input.Offset, input.Length);
-                var buffer = new byte[byteCount];
-                fixed (byte* bytes = buffer)
-                {
-                    Encoding.UTF8.GetBytes(chars + input.Offset, input.Length, bytes, byteCount);
-                }
-                var encodedName = Convert.ToBase64String(buffer);
-                return "=?utf-8?B?" + encodedName + "?=";
-            }
+            var buffer = Encoding.UTF8.GetBytes(input.Buffer);
+            var encodedName = Convert.ToBase64String(buffer);
+            return string.Concat("=?utf-8?B?", encodedName, "?=");
```

Comment:
Why Concat instead of `+`?

Reply:
At least to me it's clearer that there will only be a single string allocation. But maybe that's because I come from C++ land where `string + string + string` would result in lots of allocations by default.

Reply:
C# compiler replaces multiple string concats with string.Concat for you.